### PR TITLE
New module @turf/grid-to-matrix

### PR DESCRIPTION
**Pull request by stebogit**
@DenisCarriere I created this module to convert a `pointGrid` into a matrix of `elevation` or other `property`. I'll be working on an opposite module `matrix-to-grid`.
Both will be useful for `isobands`, `isolines` and `interpolate` I believe.
As usual I couldn't create the `README` file, and I don't really know how to properly write `types.js`.

Please let me know if you have any suggestion or comment, and feel free to change/correct anything.
Thanks!

### CODE REVIEW

File: packages/turf-grid-to-matrix/package.json [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    "@turf/meta": "^4.1.0",
+    "@turf/truncate": "^4.1.0",
+    "@turf/point-grid": "^4.1.0",
+    "benchmark": "^2.1.3",
+    "load-json-file": "^2.0.0",
+    "tape": "^4.6.3",
+    "write-json-file": "^2.0.0"
+  },
+  "dependencies": {
+    "@turf/bbox": "^4.1.0",
```
Comment (rowanwins) [inline packages/turf-grid-to-matrix/package.json]: Why is this a dependancy?

File: packages/turf-grid-to-matrix/package.json [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    "@turf/invariant": "^4.1.0",
+    "@turf/distance": "^4.1.0",
```
Comment (rowanwins) [inline packages/turf-grid-to-matrix/package.json]: why is this dependency?

File: packages/turf-grid-to-matrix/package.json [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    "@turf/helpers": "^4.1.0"
```
Comment (rowanwins) [inline packages/turf-grid-to-matrix/package.json]: Why is this is a dependency?

File: packages/turf-grid-to-matrix/package.json (hunk 1 above)
Comment (rowanwins) [inline packages/turf-grid-to-matrix/package.json]: Think you're missing @turf/meta as a dependency?

File: packages/turf-grid-to-matrix/package.json [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  "devDependencies": {
+    "@turf/meta": "^4.1.0",
+    "@turf/truncate": "^4.1.0",
+    "@turf/point-grid": "^4.1.0",
+    "benchmark": "^2.1.3",
+    "load-json-file": "^2.0.0",
+    "tape": "^4.6.3",
+    "write-json-file": "^2.0.0"
+  },
+  "dependencies": {
```
Comment (rowanwins) [inline packages/turf-grid-to-matrix/package.json]: I think `@turf/meta` is missing and you've accidently added bbox, distance and helpers

**Comment by rowanwins** [comment]
Gday @stebogit 

Dont' worry about the README files, they'll get automatically generated. And don't worry I still don't know how to write the typescript definitions files yet either (that's why @DenisCarriere is in charge!).

Nice work, although I've only had a quick look

**Comment by DenisCarriere** [comment]
👎 This module, TurfJS aims to only have GeoJSON as input & GeoJSON as output, the Matrix output seems a bit outside the scope of Turf.

However... this module does look pretty handy to have, feel free to publish this module externally and use it as a dependency for some of the Turf grid modules.

File: packages/turf-grid-to-matrix/package.json (hunk 4 above)
Comment (DenisCarriere) [inline packages/turf-grid-to-matrix/package.json]: You should actually only have `@turf/invariant` & `@turf/meta`

Comment (stebogit) [inline packages/turf-grid-to-matrix/package.json]: yes, that was a leftover from my debugging and testing

**Comment by stebogit** [comment]
I considered this (and the next `matrix-to-grid`) module/function similar to `turf-bbox` or `polygon` from `turf-helpers`, for example, where the input is actually an array. In fact my first thought was to add it to the `helpers` module, but then I realized that module has no dependencies and I thought it was on purpose.
Anyway, this means I finally have an excuse to get to learn how to publish on npm! 😃 
Should I close this PR?

**Comment by DenisCarriere** [comment]
👍 Learning how to publish on NPM will be useful :) I can help out if you have any questions, simply `@` me whenever you have a repo up and running.

If you want a very "minimalistic/robust" module/library npm template, I've published [`slippy-tile`](https://github.com/DenisCarriere/slippy-tile) & [`bbox-dateline`](https://github.com/DenisCarriere/bbox-dateline) which uses `bench` & `tape` & `rollup` (for bundling). Your library is already 90% completed, just need to glue everything together and publish via npm.

**Comment by stebogit** [comment]
[Published](https://www.npmjs.com/package/grid-to-matrix)! 🎉 😃 
Thanks @DenisCarriere!

**Comment by DenisCarriere** [comment]
👍 @stebogit Sweet!

**Pull request closed without merging**